### ISSUE TITLE
Login with school

### ISSUE THREAD

**Issue by hbh7** [PollBuddy/PollBuddy#217]
**Please describe what has to be done**
We have a register with school, so I think it would make logical sense to have the same option during register, otherwise we could get stuck with people being able to register with their school and then never being able to log in with their school lol

**Additional context**
The registration flow could probably be copied, a landing page selecting log in with Poll Buddy vs Log in with School, and then the 2 pages.

**Comment by fanga2** [comment]
For logging in on school, should that only be for RPI students? If you register with school, you're required to create a pollbuddy account anyways so logging in for school shouldn't be necessary except for RPI students.

**Comment by hbh7** [comment]
(records keeping) 
We discussed during the 7/21/2020 meeting and @fanga2 will be looking into how best to handle this. Will need to take into consideration what information CAS returns, what makes the most sense from a user usability standpoint, and a technical standpoint.

**Comment by fanga2** [comment]
Just going to leave a record so that I have something to commit. I did a bit of research and learned how to make a CAS call although it's a pretty complicated process and there's quite a bit that I don't understand myself due to not really being familiar with network stuff but I'll say what I found out. First we need to make a get request to the CAS server (the page) in order to display the login screen, then when the student inputs the password/username, we need to make a post request in order to get the ticket to the protected app aka whatever we have behind the login page. The protected app will then verify the ticket with the server (I admit I'm not exactly sure how to do this) and after that is done, the protected page should return a cookie which we then verify. I'm not sure what steps we need to implement certain things here such as the cookies and I'm not entirely sure what returning the login form actually means but I think that concludes my research at least for now.

Useful Links:
https://apereo.github.io/cas/4.2.x/protocol/CAS-Protocol.html
https://iam.harvard.edu/resources/cas-integration
https://github.com/Submitty/Submitty/issues/1803
https://apereo.github.io/cas/5.3.x/protocol/CAS-Protocol-Specification.html
